client/container_create.go

```diff
@@ -27,11 +27,18 @@ func (cli *Client) ContainerCreate(ctx context.Context, config *container.Config
 		return response, err
 	}
 
+	clientVersion := cli.ClientVersion()
+
 	// When using API 1.24 and under, the client is responsible for removing the container
-	if hostConfig != nil && versions.LessThan(cli.ClientVersion(), "1.25") {
+	if hostConfig != nil && versions.LessThan(clientVersion, "1.25") {
 		hostConfig.AutoRemove = false
 	}
 
+	// When using API under 1.42, the Linux daemon doesn't respect the ConsoleSize
+	if hostConfig != nil && platform != nil && platform.OS == "linux" && versions.LessThan(clientVersion, "1.42") {
```

Comment:
by my comment on endpoint, I don't think we should have the client manage default values to be set.

Reply:
See my other comment; effectively this is "resetting" the value back to the default for the type https://github.com/moby/moby/pull/43593/files#r876720241

(I guess `[2]int{}` would've worked as well if you think that's clearer though)